### PR TITLE
[dapps] adds kiosk-cli dapp

### PR DESCRIPTION
**Pull request by damirka**
## Description 

Adds the `kiosk-cli` demo application to the `dapps/` folder.

## Test Plan 

Extensively tested on testnet.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes

- none

### CODE REVIEW

File: dapps/kiosk-cli/index.js [hunk 1]
```diff
@@ -0,0 +1,841 @@
...
+    txb,
+    itemInfo.data.type,
+    { price },
+    kioskId,
+    itemInfo.data.objectId,
+    policies[0],
+  );
+
+  // For the locking policy scenario when an item needs to be locked;
+  if (item === null) {
```
Comment (manolisliolios) [inline dapps/kiosk-cli/index.js:581]: This has changed on 0.3.0 version.
`const { item, canTransfer } = purchaseAndResolvePolicies()`

File: dapps/kiosk-cli/index.js [hunk 2]
```diff
@@ -0,0 +1,847 @@
...
+async function findKioskCap(address) {
+  const sender = address || (await signer.getAddress());
+
+  if (!isValidSuiAddress(sender)) {
+    throw new Error(`Invalid address "${sender}"`);
+  }
+
+  const objects = await provider.getOwnedObjects({
+    owner: sender,
+    filter: { StructType: '0x2::kiosk::KioskOwnerCap' },
```
Comment (manolisliolios) [inline dapps/kiosk-cli/index.js:748]: maybe use the KIOSK_OWNER_CAP type we have for better reusability

File: dapps/kiosk-cli/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "kiosk-sdk",
```
Comment (Jordan-Mysten) [inline dapps/kiosk-cli/package.json:2]: name seems wrong

File: dapps/kiosk-cli/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+build
```
Comment (Jordan-Mysten) [inline dapps/kiosk-cli/.gitignore:1]: Do you actually need this file

**Review by Jordan-Mysten — COMMENTED** [review]
No real concerns, just not sure why we need a CLI given we have a web app. Also might make more sense to put this into the `sui` CLI?

File: dapps/kiosk-cli/index.js [hunk 5]
```diff
@@ -0,0 +1,849 @@
...
+ * Find the KioskOwnerCap at the sender address.
+ */
+async function findKioskCap(address) {
+  const sender = address || (await signer.getAddress());
+
+  if (!isValidSuiAddress(sender)) {
+    throw new Error(`Invalid address "${sender}"`);
+  }
+
+  const objects = await provider.getOwnedObjects({
```
Comment (manolisliolios) [inline dapps/kiosk-cli/index.js:748]: Can we rewrite this to use the `getOwnedKiosks` function from the SDK?

**Review by manolisliolios — APPROVED** [review]
LGTM!

**Pull request merged**